feat: Migration guide (#289)
paritytech/substrate-api-sidecar: MIGRATION_GUIDE.md

# v0.x.x &rightarrow; v1.0.0-beta.0

Full documentation for v1 endpoints is available here: https://paritytech.github.io/substrate-api-sidecar/dist/

For most users, the major bump will require transitioning all the endpoint paths

> they use to an updated version. The v1.0.0.beta.0 release contains both the old

Reviewer: ```suggestion
that they use to an updated version. The v1.0.0.beta.0 release contains both the old
```

and new endpoints to help facilitate the transition.

> **`substrate-api-sidecar` releases after Friday, October 2nd, 2020, will not

Reviewer: This might be a little aggressive on time. Maybe do one more week.

Author: See updated. I moved it back to October 16th.

make any gurantees to include deprecated endpoints.** It is important that users

> finishing transitioning to the new endpoints by this date so they are ready for

Reviewer 2: ```suggestion
complete the transition to the new endpoints by this date so they are ready for
```

any subsequent emergency updates.

> Below is a table that is intended to give an overview of what the mapping of
> deprecated endpoints to there v1 equivalents. Many of the new

Reviewer: ```suggestion
Below is a table that is intended to give an overview of the mapping from
deprecated endpoints to their v1 equivalents. Many of the new
```

> Below is a table that is intended to give an overview of what the mapping of
> deprecated endpoints to there v1 equivalents. Many of the new

Reviewer 2: ```suggestion
Below is a table that gives an overview of what the mapping from
deprecated endpoints to their v1 equivalents. Many of the new
```

> endpoints have unlisted options for query params.

Reviewer: Add a note that some of the response objects are different too. Or maybe add a symbol on the v1 side of the table, like `*` to indicate that the response has a breaking change.

Edit: Oh, you cover it after

Reviewer 2: "unlisted"? Can you elaborate on what that means?

Author: I removed "unlisted" and rephrased it to basically say "go look at the docs for more info on query params".

| (None)                            | `/transaction/dry-run`                            |
| `/metadata`                       | `/runtime/metadata`                               |
| `/metadata/{blockId}`             | `/runtime/metadata?at={blockId}`                  |
| (None)                            | `/runtime/spec`                                   |
| (None)                            | `runtime/code`                                    |
| `/staking-info`                   | `/pallets/staking/progress`                       |
| `/staking-info/{blockId}`         | `/pallets/staking/progress?at={blockId}`          |
| `/staking-info/{blockId}`         | `/pallets/staking/progress?at={blockId}`          |
| (None)                            | `/pallets/{palletId}/storage/{storageItemId}`     |
| (None)                            | `/node/network`                                   |
| (None)                            | `/node/version`                                   |
| (None)                            | `/node/transaction-pool`                          |
| `/`                               | `/`

**N.B.** the `/blocks` endpoint sees two significant breaking changes:

> 1) extrinsic and event methods names no longer are a string of the form

Reviewer: ```suggestion
1) extrinsic and event method names are no longer a string of the form
```

> 1) extrinsic and event method names are no longer a string of the form
> `pallet.method`, instead they are an object of the form

Reviewer: ```suggestion
1) Extrinsic and event method names are no longer a string of the form
`pallet.method`; instead they are an object of the form
```

`{ pallet: string, method: string }`.

2) Unsigned extrinsics and inherents will have `null` for nonce and tip, while

> `paysFee` will always be `false`. Previously the latter two fields where 0, while

Reviewer: ```suggestion
`paysFee` will always be `false`. Previously the latter two fields were 0, while
```

`paysFee` was often `true`. (#274)

There are some other changes throughout the API, including some slightly updated

> error messages which have not yet been updated in the new docs UI.

Reviewer: ```suggestion
error messages that have not yet been updated in the new docs UI.
```
